### PR TITLE
feat(agent-manager): add fixed local session tab with branch display and persistence

### PR DESCRIPTION
**Pull request by marius-kilocode**
## Summary

- Add a dedicated **local session** item at the top of the agent manager sidebar that always runs in the main repo (never a worktree)
- The local session persists across panel reopens via VS Code webview state API (`getState`/`setState`)
- Shows the current git branch name, refreshed on click via `agentManager.requestRepoInfo` message
- Keyboard navigation (Cmd+Up/Down) includes the local item, with extracted pure `resolveNavigation()` function and 16 unit tests
- Sessions created from the local tab stay bound to it and are filtered from the SESSIONS list
- Styling matches session row design (not a card) for visual consistency

### Extension side (`AgentManagerProvider.ts`)
- `sendRepoInfo()` reads current branch via `git rev-parse --abbrev-ref HEAD` and pushes to webview
- Called on panel open and on `agentManager.requestRepoInfo` webview message
- `WorktreeManager.currentBranch()` visibility changed from private to public

### Webview side (`AgentManagerApp.tsx`)
- `localSessionID` signal with webview state persistence for recovery
- `onLocal` signal tracks whether user is viewing the local workspace
- `sessionCreated` message listener adopts new sessions as local when created from local view
- `selectLocal()` requests fresh branch info on every click
- `isLocal()` derived check drives sidebar highlight and session list filtering

### Navigation (`navigate.ts` + `navigate.test.ts`)
- Pure `resolveNavigation()` function handles local (index -1) + session list (0..N-1)
- 16 tests covering all boundary conditions, unknown session IDs, and full walk-through

### Types (`messages.ts`)
- `AgentManagerRepoInfoMessage` (extension → webview)
- `RequestRepoInfoMessage` (webview → extension)

### Styles (`agent-manager.css`)
- Local item styled like a session row (matching padding, border-radius, hover/active states)
- Session items use softer non-contrasting active state

image
image

### CODE REVIEW

File: packages/kilo-vscode/webview-ui/agent-manager/AgentManagerApp.tsx [hunk 1]
```diff
@@ -49,19 +51,52 @@ const AgentManagerContent: Component = () => {
...
+    [...session.sessions()]
+      .filter((s) => s.id !== localSessionID())
+      .sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()),
   )
 
+  const selectLocal = () => {
+    setOnLocal(true)
+    // Request fresh branch info — cheap git rev-parse, ensures branch name
+    // is current even if the user switched branches in an external terminal
+    vscode.postMessage({ type: "agentManager.requestRepoInfo" })
```
Comment (markijbema) [inline packages/kilo-vscode/webview-ui/agent-manager/AgentManagerApp.tsx:82]: as a general comment: do we want to consider having some sort of git-logic system and have that as a tool agent manager uses, instead of it being a part of agent manager?

and then: should that git tooling be in the core, so we can also have an agent manager in jetbrains? or as a standalone?

File: packages/kilo-vscode/webview-ui/agent-manager/AgentManagerApp.tsx [hunk 2]
```diff
@@ -103,33 +155,87 @@ const AgentManagerContent: Component = () => {
...
+        >
           + New Agent
         </Button>
+        <button class={`am-local-item ${isLocal() ? "am-local-item-active" : ""}`} onClick={() => selectLocal()}>
+          <svg class="am-local-icon" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <rect x="2.5" y="3.5" width="15" height="10" rx="1" stroke="currentColor" />
+            <path d="M6 16.5H14" stroke="currentColor" stroke-linecap="square" />
+            <path d="M10 13.5V16.5" stroke="currentColor" />
+          </svg>
+          <div class="am-local-text">
```
Comment (markijbema) [inline packages/kilo-vscode/webview-ui/agent-manager/AgentManagerApp.tsx:220]: as discussed irl, we probably want to use components more

Comment (marius-kilocode) [reply packages/kilo-vscode/webview-ui/agent-manager/AgentManagerApp.tsx:220]: Will redesign in follow up.

File: packages/kilo-vscode/webview-ui/agent-manager/AgentManagerApp.tsx (hunk 1 above)
Comment (marius-kilocode) [reply packages/kilo-vscode/webview-ui/agent-manager/AgentManagerApp.tsx:82]: Yes, I will later extract it to core
